### DISCUSSION TITLE
Feature Request - Set User Volume in Server XML Config for TeamTalk5 Server

### DISCUSSION THREAD

**Discussion by AgentSmith2001** [BearWare/TeamTalk5#1821] [Ideas]
### Description

The current version of the TeamTalk5 server does not have an option for admins to set user volume levels in the server XML configuration file. This feature request is to add a setting that allows admins to set a default volume level for individual users when they log into the server.
The proposed feature would provide two significant benefits. Firstly, it would prevent users from raising their volume levels beyond what the admin permits, thus avoiding loud and disruptive noises. Secondly, it would benefit those using Samsung devices with low volume levels even at 100% gain by allowing them to have a more acceptable volume level.
This feature request is similar to the "Fixed volume for all users" setting available in the channel settings. However, the proposed feature provides more granular control over individual user volume levels.
Implementing this feature would provide admins with greater control and customization options, thereby improving the overall user experience. Users logging into the server would have a consistent and appropriate volume level, preventing any unpleasant surprises or disruptions.
Thank you for considering this feature request.

### Application

- [X] qtTeamTalk
- [ ] TeamTalkAndroid
- [ ] iTeamTalk
- [X] TeamTalkServer

**Comment by amirmahdifard** [comment]
yeah i agree. There should be more user rigths for admins to controle users, including preventing them from sending private messages to people and sending channel messages that i've requested before and developers just closed it

**Comment by bear101** [comment]
I believe using "Audio Configuration" in Channel dialog is the correct way to solve this issue.

**Reply by AgentSmith2001** [reply]
This applies to all users on the channel, rather I need a feature to control the audio volume for each user individually on every channel.

**Reply by muamel1** [reply]
And Also media file volume

**Comment by ghost** [comment]
I am also interested this feature to be implemented

**Comment by aliRahimi1997** [comment]
As an administrator on my server, I'd also like this feature to be implemented, because sometimes its really hard to control the audio volume for users, and they become noisy when they set their microphone gain to 100%. This feature must be added in tt5server xml config file so administrators can control audio volume for each user individually. It must stay affectivve even after logging out and re-login again. It must be available in user accounts dialog under each user preferences in codec limitations tab

**Comment by AmiraliKhatami459** [comment]
Hey! Thanks for creating such discussion and requesting this feature! Yes, I am agree as well as others who they've commented here. Adding a feature in team talk 5 server xml config which provides the
capability of control audio volume for each user individually would be
a great idea. it would prevent users from raising their volume levels
beyond what the admin permits, thus avoiding loud and disruptive
noises. This feature exists in channel properties dialog, but I need
such feature in user accounts dialog as the moderator on my server. Sometimes its really hard to control people specially when teaching in online classes and students are new to TeamTalk. I hope this feature gets added in 5.13 or newer versions. Thanks for reading my comments dear developers! Your tries are seriously valued. Thanks for the efforts for the improvements of team talk, whether default qt, whether IOS or the tt5 server

**Comment by bear101** [comment]
Have you tried out the "Restore volume settings and subscriptions" in General-tab in "Preferences dialog". Then each user have their own stored volume settings. You can try it on the official "TeamTalk 5 Encrypted Server".

**Reply by AmiraliKhatami459** [reply]
Hey! Thanks for the answer. First, this option is only available for those servers which is made using bearware.dk web login, most of servers we use does not support this option since they are private. 
Second, As the name suggests, this option is not for adjust audio volume for each user. It must be available to server administrators only. They must have the ability to have full control on abusing people, including adjusting audio volume for users individually (If required). The option you mentioned is for tt5 client, not tt5server. 
If you need more explanation, please ask so I can say more. Its really simple. It must work as same as the abuse prevention (Flood protection), or codec limitation in user accounts dialog. Thanks in advance!

**Reply by AgentSmith2001** [reply]
It doesn't work. Because its for web login users. It is defined on the client, not on the server. The best solution is add "Enable fixed audio volume for this user" in user accounts dialog for each user

**Comment by aliRahimi1997** [comment]
It doesn't work on private servers. It works on web login servers only. This will not give ability to control audio volume for an admin

**Comment by marlene-1992** [comment]
Hmm, fully agreed. I upvoted this suggestion. There must be more customizations in team talk like the proposed feature, and admins should have more rights to monitor people. Default QT is on the good way of Development, I hope this applies to tt5 server too, and more features gets added. 
I have personally seen that many times people make crowd noises and their audio volume is beyond what it should be, especially in private and non-encrypted servers. The proposed feature will give an ability to server admins to control the loudness of each user.
I am a teacher. I have personally experienced that being an administrator is sometimes seriously needed, to stop transmitting voice/channel text messages when people who doesn't use headphone are simultaneously talking together or spamming on the channel, e.g. But implementing a feature to control the loudness of each user would be a great idea and will give more customization to TT5 server.

**Comment by kempfious619** [comment]
@bear101 
Yes of course. Why not? Do not hesitate to add this feature. "Audio Configuration" in Channel dialog is definitely not enough because each channel is intended to have  different configuration, bitrate, and sample rate.

**Comment by Muhsine-t** [comment]
@bear101 
I'd like to control audio volume of users on my server. I kindly ask team talk developer to implement this feature as a part of codec limitation or abuse prevention. Its not good idea to control audio volume from the channel properties due to differences between different soundcards, but its good idea to control it individually for each user.
